data/XML/vocations.xml

```diff
@@ -1,6 +1,9 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <vocations>
 	<vocation id="0" clientid="0" name="None" description="none" gaincap="10" gainhp="5" gainmana="5" gainhpticks="6" gainhpamount="1" gainmanaticks="6" gainmanaamount="1" manamultiplier="4.0" attackspeed="2000" basespeed="220" soulmax="100" gainsoulticks="120" fromvoc="0">
+		<flags>
+			<flag allowPvp="0" />
+		</flags>
```

Comment:
I don't think it makes sense to have this parent-child structure of "list of flags", where each flag contains attributes... and it can actually contain many such attributes, because why not?

Specifically, the design here does not prevent this:
```xml
<flags>
	<flag allowPvp="0" allowSomething="1" />
	<flag allowSomethingElse="0" />
</flags>
```

There's really not good way to avoid that, because this (below) is less flexible, although does solve this problem somewhat:
```xml
<flags>
	<flag>allowPvp</flag>
	<flag>allowSomethingElse</flag>
</flags>
```

The problem is that everything would be a boolean now, and can't be defaulted to true if not specified.

So I think the simplest way to do that is less nesting:
```xml
<flags allowPvp="0" something="1" somethingElse="1" />
```

Reply:
Well, you approved it for groups too, which uses the same design https://github.com/otland/forgottenserver/pull/2546

Reply:
IMHO it's better to be consistent with what's there instead of introducing new design for every new feature.

With that said, I could make it consistent with what's already in vocations, skills, so it would be repeated `flag` nodes, without nesting:
```xml
<flag name/id="allowPvp" value="1" />
```

Or keep it consistent with other flag declarations in monsters and groups, and leave it as is.

Reply:
Anyway, I don't see whats ambiguous. Your example:
```xml
<flags>
	<flag allowPvp="0" allowSomething="1" />
	<flag allowSomethingElse="0" />
</flags>
```

Is not a valid use case, `allowSomething` will be ignored as the flag takes _only_ the first attribute.

If you really wanna do something new here, then maybe:
```xml
<flags>
  <allowPvp />
  <!-- which is equivalent to -->
  <allowPvp>1</allowPvp>
</flags>
```

Defining a flag without value makes it "set" (true) by default.

Anyway, I'm not even sure about the `flags` name, since I might want to add explicit `protectionLevel` per vocation, which would not really fit into `flags`. Maybe `attributes`? Or `features`?
```xml
<features>
  <pvp>0</pvp>
  <lootDrop>1</lootDrop>
  <protectionLevel>40</protectionLevel>
  <skillLossReduction>10</skillLossReduction>
</features>
```

🤷

Reply:
Oh my god XML is so damn ugly

Reply:
> Is not a valid use case, allowSomething will be ignored as the flag takes only the first attribute.

That's exactly the ambiguity problem. It's not obvious to the user. What you're describing is an implementation limitation.

We all know XML design is a problem and #2546 showed a similar problem. If you want to follow that other style for consistency, that is fine as well. I think what made the difference there is the vast number of flags, where here we only have one for now?

On the other hand, if you call it features, which really turn into parameters... then why call it anything at all? Everything inside vocation definition is a parameter/property/feature/flag/option... whatever. Maybe just keep adding extra attributes to the top level `<vocation/>`? Ugly, but it already is anyway, and simple. You're right that these flags aren't really distinctive "flags".

Reply:
> That's exactly the ambiguity problem. It's not obvious to the user. What you're describing is an implementation limitation.

It's not really ambiguous with current implementation... You have a nested single flag under flags, so why would you think you can assign more than one flag attribute to flag node? 🤔 
It's kinda really explicit, flags collection containing entries describing single flag.

It seems like you'd be most comfortable with how skills are defined, full example here:
```xml
<vocation id="1" ...>
	<formula meleeDamage="1.0" distDamage="1.0" defense="1.0" armor="1.0" />
	<flag id="allowPvp" value="0" />
	<skill id="0" multiplier="1.5" />
	<skill id="1" multiplier="2.0" />
	<skill id="2" multiplier="2.0" />
</vocation>
```

Flattening flags might be actually a good idea, as in your own example:
```xml
<flags allowPvp="0" something="1" somethingElse="1" />
```

But we might want to resolve first how do we want to call it, if these are not gonna be only simple boolean flags. Personally I would rather not add more attributes to the root node (vocation).

Reply:
We already have flags that work exactly like this for monsters too. I don't see a problem being consistent with what we already have implemented. Not all of monster flags are boolean. Some are numeric like targetdistance and staticattack. Seems better off to add it as is and look at it again in 2.0 if that is the timeframe for really switching away from XML.

Reply:
I support the idea of ​​squashing it into one line:
`<flags allowPvp="0" something="1" somethingElse="1" />`
is consistent with what we already have in this xml file